File: back-end/models/Lecturematerial.js
import mongoose from "mongoose";
const { Schema } = mongoose;

const lecturematerialSchema = new Schema(
  {
    lesson: {
      type: String,
      required: true,
    },
    Teachersubject: {
      type: String,
      required: true,
    },
    Tmedium: {
      type: String,
      required: true,
    },
    TeacherEmail: {
      type: String,
      required: true,
    },
    zoom: {
      type: String,
      required: true,
    },
    PDF: {
      type: String,
      required: true,
    },

    video: {
      type: String,
    },
    otherlink: {
      type: String,
    },
  },
  {
    timestamps: true,
  }
);
export default mongoose.model("lecturematerial", lecturematerialSchema);

File: dream_learn_app/dream_learn_app/lib/models/lecture_material.dart
class LectureMaterial{
  String? lessonName;
  String? pdfPath;
  Uri? zoomLink;
  Uri? lectureRecording;
  Uri? otherLinks;

  LectureMaterial.fromJson(Map<String,dynamic> json){
  lessonName=  json['lesson'];
  pdfPath=  json['PDF'];
   zoomLink= Uri.tryParse(json['zoom']
   
   
   );
   lectureRecording = Uri.tryParse(json['video']);
    otherLinks = Uri.tryParse(json['otherlink']);
  }

}